### PR TITLE
Render labels as HTML

### LINKED ISSUE #5623: Minor, use HTML for input labels
Currently all HTML tags in input labels are handled as raw text. However, there are significantly more cases where we'd want to be able to use HTML tags in input labels than the other way around. Examples: adding a link, a help dialog icon, some simple styling, etc.

### PR DESCRIPTION
**Pull request by azjps**
There's no need for raw text labels for form elements here, afaik. On
the other hand there are a lot of reasonable reasons to want to use HTML
tags in labels, for example adding a link or a help dialog widget.

All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [x] issues: fixes #5623
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -15,3 +15,4 @@ prune examples/compat/seaborn/*.html
 prune examples/embed/*.html
 prune examples/glyphs/*.html
 prune examples/plotting/file/*.html
+include bokeh/_version.py
```
Comment (mattpap) [inline MANIFEST.in]: This change doesn't seem related.

File: examples/models/widgets.py [hunk 2]
```diff
@@ -39,15 +39,15 @@
 
 radio_button_group = RadioButtonGroup(labels=["Option 1", "Option 2", "Option 3"], active=0)
 
-text_input = TextInput(placeholder="Enter value ...")
+text_input = TextInput(placeholder="Enter value ...", title="<b>text</b>")
```
Comment (mattpap) [inline examples/models/widgets.py]: Users may depend on the fact that `title` used to accept raw strings. This is a breaking change. A bigger concern, however, is mixing raw strings with HTML, which never is a good idea (refer to PHP and a ton of other language/frameworks to see how this works out). Raw strings and HTML have to be separated to avoid user confusion, subtle bugs and security concerns. Unfortunately, we already broke this rule in a few places ourselves. We need to fix this across bokeh. Here, I would go with `TextInput(..., title=HTML("<b>text</b>"))`, where `HTML` is a model or a struct (raw `HasProps`), and leave the default for `title` to be raw string (so the type would be `title = Either(String, HTML)`).

File: MANIFEST.in (hunk 1 above)
Comment (azjps) [inline MANIFEST.in]: Yeah, more git ui issues.

File: examples/models/widgets.py (hunk 2 above)
Comment (azjps) [inline examples/models/widgets.py]: That makes sense, definitely bigger scope of a change than what I was expecting to do though. `HTML` could inherit from `Markup` abstract model perhaps?

Comment (mattpap) [inline examples/models/widgets.py]: > HTML could inherit from Markup abstract model perhaps?

That's a possibility, but I have to give this a broader though.

Comment (azjps) [inline examples/models/widgets.py]: Sure, let me know -- I'll just use this changeset locally until then.

**Comment by bryevdv** [comment]
@azjps apologies this is taking so long. The original issue was not tagged, which made it harder to keep track of. I have tagged marked it for `0.12.7` though as @mattpap mentions we may want to consider a different approach than sending HTML as raw text. Because of that, and since there are now also conflicts, I am closing this PR. If you'd like to discuss approaches for a new PR, let's continue in the issue.
